src/client.rs

```diff
@@ -280,3 +284,416 @@ where
         Ok(signed)
     }
 }
+
+/// This struct represents a subscription to the progress of some transaction, and is
```

Comment:
I reckon all this stuff deserves it's own file, shouldn't be in `client.rs`

Reply:
I actually did do this initially. I like the separation; I just wasn't keen on the circular dependency it would introduce (client makes a transactionprogress, transactionprogress et al have a reference to client) :)

Reply:
(I think I'll give it a try anyway though, because it would be nice breaking that file down!)